osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs

```diff
@@ -192,6 +201,15 @@ protected override void LoadComplete()
             refresh();
         }
 
+        protected override void UpdateAfterChildren()
+        {
+            base.UpdateAfterChildren();
+
+            // best effort to confine the auto-sized text to parent bounds
+            // TODO: remove when text/link flow can support truncation with ellipsis natively.
+            beatmapText.MaxWidth = mainFillFlow.DrawWidth;
```

Comment:
Can this sort of thing go live as a default in `TruncatingSpriteText`? Sprinkling it everywhere that we've got this class already seems... like a waste of effort.

I know the wedge usage is slightly different but maybe it can be padded differently to achieve the same effect?

Reply:
I did some normalising when I was implementing the [truncated text tooltips](https://github.com/ppy/osu/pull/23829) (also with `RelativeSizeAxes`).
> I also tried centralising the `MaxWidth` and `RelativeSizeAxes = Axes.X` assigning to `TruncatingSpriteText` (https://github.com/Joehuu/osu/commit/dd8e1ba68d3a7fb46b128b6268446929f1e01e61), but not sure about the solution as I'm working around a framework limitation and probably should be done more directly in `SpriteText`.

But now links are in the mix, and here's what my updated solution is (using `ChildSize.X` and omited `X` and `RelativeSizeAxes` for now): https://github.com/ppy/osu/commit/e45a2b2ff14d50fd1c43d5bab723d088948b2d53

May need to incorporate the `RelativeSizeAxes` removal as it makes `MaxWidth` a relative value.

Reply:
What is even the framework limitation being worked around here?

Reply:
Not sure what I was saying at the time, but probably putting said logic framework-side in a proper way (through the use of properties) so framework consumers can use it and there is no usage of `Update()` game-side, but probably not feasible at the moment.